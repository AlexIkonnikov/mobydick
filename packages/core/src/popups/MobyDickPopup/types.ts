import {FC} from 'react';

import {IPopupsContext} from '../context';
import {IContentProps, IPopup} from '../types';

export type ModalProps = Partial<Omit<IContentProps, 'onClose'>> & {
  onClose?(id: string): void;
};

export interface IPopupWithProps extends IPopup {
  props: Record<string, unknown> & ModalProps;
}

export interface IModalRef extends Pick<IPopupsContext, 'closeAllPopups'> {
  openPopup<Props>(
    Component: FC<Props & IContentProps>,
    props?: Props & ModalProps,
  ): void;
  closePopup(id: string): void;
}
