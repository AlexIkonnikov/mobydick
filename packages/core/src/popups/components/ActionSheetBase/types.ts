import {StyleProp, ViewStyle} from 'react-native';
import React, {PropsWithChildren, ReactElement} from 'react';

import {TypographyProp} from '../../../typography/types';
import {IPopup} from '../../types';

export interface IPropsContents {
  title: string;
  textFont?: TypographyProp;
  leftIcon?: ReactElement | undefined;
  onPress?(): void;
  radio?: string;
  checkboxList?: string[];
  disabled?: boolean;
  itemType: IItemType;
}

export enum IItemType {
  firstItem = 'firstItem',
  innerItem = 'innerItem',
  lastItem = 'lastItem',
  singleItem = 'singleItem',
  cancelItem = 'cancelItem',
}

export interface IPropsItem extends IPropsContents {
  style?: ViewStyle;
  isStatusPressedForTest?: boolean;
}

export interface IActionSheetBaseProps
  extends PropsWithChildren<Omit<IPopup, 'Content'>> {
  onClose: () => void;
  overlayStyle?: StyleProp<ViewStyle>;
  containerStyle?: StyleProp<ViewStyle>;
  children: React.ReactNode;
}
