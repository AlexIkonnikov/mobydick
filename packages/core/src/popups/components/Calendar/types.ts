export interface IChangeDate {
  dateStart: string;
  dateEnd: string;
}

export enum IButtonView {
  small = 'small',
  large = 'large',
}
