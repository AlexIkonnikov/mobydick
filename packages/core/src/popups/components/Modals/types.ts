import {IButtonTypes} from '../../../cta';

export interface IModalProps {
  title: string;
  descriptionText: string;
  buttonText?: string;
}

export interface IModalAskProps {
  title: string;
  descriptionText: string;

  typeLeft?: IButtonTypes;
  textLeft?: string;

  onPressRight(): void;
  typeRight?: IButtonTypes;
  textRight?: string;
}
