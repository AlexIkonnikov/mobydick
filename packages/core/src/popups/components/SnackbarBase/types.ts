import {StyleProp, ViewStyle} from 'react-native';
import {PropsWithChildren} from 'react';

import {IPopup, IPosition} from '../../types';

export interface ISnackbarProps
  extends PropsWithChildren<Omit<IPopup, 'Content'>> {
  onClose: () => void;
  containerStyle?: StyleProp<ViewStyle>;
  position: IPosition;
  timeShow?: number;
}
