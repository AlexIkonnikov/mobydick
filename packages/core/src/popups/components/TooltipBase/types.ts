import {StyleProp, ViewStyle} from 'react-native';
import {PropsWithChildren, RefObject} from 'react';

import {IPopup, IPosition} from '../../types';
import {ITouchableOpacity} from '../../../basic';

export enum IPlacement {
  start = 'start',
  center = 'center',
  end = 'end',
}

export interface IArrowViewPopup {
  placement: IPlacement;
  position: IPosition;
  arrowViewStyles?: ViewStyle | ViewStyle[];
}

export interface ITooltipBaseProps
  extends PropsWithChildren<Omit<IPopup, 'Content'>> {
  onClose: () => void;
  position: IPosition;
  placement: IPlacement;
  refCurrent: RefObject<ITouchableOpacity>;
  overlayStyle?: StyleProp<ViewStyle>;
  containerStyle?: StyleProp<ViewStyle>;
  timeShow?: number;
}
