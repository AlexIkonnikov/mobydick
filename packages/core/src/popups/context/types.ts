import {IPopup, IPopupId} from '../types';

export type IOpenPopupParams = Partial<IPopup> & Pick<IPopup, 'Content'>;

export interface IPopupsContext {
  popups: IPopup[];
  openPopup: (popup: IOpenPopupParams) => string;
  closePopup: (id: IPopupId) => void;
  closeAllPopups: () => void;
}
