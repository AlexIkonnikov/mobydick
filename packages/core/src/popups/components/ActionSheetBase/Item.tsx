import {FC, useCallback} from 'react';

import useStyles from '../../../styles/hooks/useStyles';
import useTheme from '../../../styles/hooks/useTheme';
import Pressable from '../../../basic/components/Pressable/Pressable';
import {LABELS} from '../../../other';

import stylesCreate from './stylesCreate';
import Contents from './content/Contents';
import {IPropsItem} from './types';

const Item: FC<IPropsItem> = props => {
  const {onPress, style, disabled, itemType, isStatusPressedForTest} = props;
  const [styles] = useStyles(stylesCreate, itemType);
  const {colors} = useTheme();

  const getStyle = useCallback(
    ({pressed}: {pressed: boolean}) => [
      styles.item,
      {backgroundColor: pressed ? colors.BgSecondary : colors.BgPrimary},
      style,
    ],
    [],
  );

  return (
    <Pressable
      style={getStyle}
      disabled={disabled}
      onPress={onPress}
      accessibilityLabel={LABELS.actionSheetsItem}
      testOnly_pressed={isStatusPressedForTest}>
      <Contents {...props} />
    </Pressable>
  );
};

export default Item;
