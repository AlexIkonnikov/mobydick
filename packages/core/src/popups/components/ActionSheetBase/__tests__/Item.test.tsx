import {fireEvent, render} from '@testing-library/react-native';
import React from 'react';

import Item from '../Item';
import {IItemType} from '../types';
import Constants from '../../PopupBase/constants';
import SimpleIcon from '../../../../styles/icons/font/SimpleIcon';

describe('@npm/mobydick-core/ActionSheetBase', () => {
  it('should renders correctly Item', () => {
    const {toJSON} = render(
      <Item title={'title'} itemType={IItemType.cancelItem} />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('should renders correctly Item with  textFont', () => {
    const {toJSON} = render(
      <Item
        textFont={'Regular-White-S'}
        itemType={IItemType.firstItem}
        title={'title'}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('should renders correctly inner item', () => {
    const {toJSON} = render(
      <Item
        textFont={'Regular-White-S'}
        itemType={IItemType.innerItem}
        title={'title'}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('should renders correctly Item with !selected', () => {
    const {toJSON} = render(
      <Item
        itemType={IItemType.lastItem}
        radio={'selected'}
        onPress={() => null}
        textFont={'Regular-White-S'}
        title={'title'}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('should renders correctly Item with selected', () => {
    const {toJSON} = render(
      <Item
        itemType={IItemType.singleItem}
        checkboxList={['title']}
        onPress={() => null}
        title={'title'}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('should renders correctly Item with disabled', () => {
    const {toJSON} = render(
      <Item disabled={true} itemType={IItemType.cancelItem} title={'title'} />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('should renders correctly Item with leftIcon', () => {
    const {toJSON, getByLabelText} = render(
      <Item
        itemType={IItemType.cancelItem}
        title={'title'}
        leftIcon={<SimpleIcon name={'icon-copy'} />}
        isStatusPressedForTest
      />,
    );
    const pressableItem = getByLabelText(
      Constants.accessibilityLabelActionSheetsItem,
    );
    fireEvent.press(pressableItem);

    expect(toJSON()).toMatchSnapshot();
  });
});
