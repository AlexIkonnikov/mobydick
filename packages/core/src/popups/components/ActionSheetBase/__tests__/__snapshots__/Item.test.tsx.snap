// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@npm/mobydick-core/ActionSheetBase should renders correctly Item 1`] = `
<View
  accessibilityLabel="accessibilityLabelActionSheetsItem"
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 36,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 90,
        "marginHorizontal": 24,
        "minHeight": 150,
        "paddingHorizontal": 60,
      },
      {
        "backgroundColor": "#FFF",
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <Text
      allowFontScaling={true}
      style={
        [
          {
            "color": "#2B78EE",
            "fontFamily": "Inter-Medium",
            "fontSize": 48,
            "lineHeight": 60,
          },
          {
            "paddingVertical": 45,
          },
        ]
      }
    >
      title
    </Text>
  </View>
</View>
`;

exports[`@npm/mobydick-core/ActionSheetBase should renders correctly Item with  textFont 1`] = `
<View
  accessibilityLabel="accessibilityLabelActionSheetsItem"
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderTopLeftRadius": 36,
        "borderTopRightRadius": 36,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": 24,
        "minHeight": 150,
        "paddingHorizontal": 60,
      },
      {
        "backgroundColor": "#FFF",
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <Text
      allowFontScaling={true}
      style={
        [
          {
            "color": "#FFF",
            "fontFamily": "Inter-Regular",
            "fontSize": 45,
            "lineHeight": 57,
          },
          {
            "paddingVertical": 45,
          },
        ]
      }
    >
      title
    </Text>
  </View>
</View>
`;

exports[`@npm/mobydick-core/ActionSheetBase should renders correctly Item with !selected 1`] = `
<View
  accessibilityLabel="accessibilityLabelActionSheetsItem"
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderBottomLeftRadius": 36,
        "borderBottomRightRadius": 36,
        "borderTopColor": "#D1D4DB",
        "borderTopWidth": 3,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 24,
        "marginHorizontal": 24,
        "minHeight": 150,
        "paddingHorizontal": 60,
      },
      {
        "backgroundColor": "#FFF",
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <Text
      allowFontScaling={true}
      style={
        [
          {
            "color": "#FFF",
            "fontFamily": "Inter-Regular",
            "fontSize": 45,
            "lineHeight": 57,
          },
          {
            "paddingVertical": 45,
          },
        ]
      }
    >
      title
    </Text>
  </View>
  <View
    needsOffscreenAlphaCompositing={true}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "opacity": 1,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityState={
        {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "aspectRatio": 1,
          "backgroundColor": "transparent",
          "borderColor": "rgba(32, 36, 45, 0.3)",
          "borderRadius": 33,
          "borderWidth": 6,
          "justifyContent": "center",
          "width": 66,
        }
      }
      value="title"
    />
  </View>
</View>
`;

exports[`@npm/mobydick-core/ActionSheetBase should renders correctly Item with disabled 1`] = `
<View
  accessibilityLabel="accessibilityLabelActionSheetsItem"
  accessibilityState={
    {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 36,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 90,
        "marginHorizontal": 24,
        "minHeight": 150,
        "paddingHorizontal": 60,
      },
      {
        "backgroundColor": "#FFF",
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <Text
      allowFontScaling={true}
      style={
        [
          {
            "color": "#9BA1B0",
            "fontFamily": "Inter-Medium",
            "fontSize": 48,
            "lineHeight": 60,
          },
          {
            "paddingVertical": 45,
          },
        ]
      }
    >
      title
    </Text>
  </View>
</View>
`;

exports[`@npm/mobydick-core/ActionSheetBase should renders correctly Item with leftIcon 1`] = `
<View
  accessibilityLabel="accessibilityLabelActionSheetsItem"
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 36,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 90,
        "marginHorizontal": 24,
        "minHeight": 150,
        "paddingHorizontal": 60,
      },
      {
        "backgroundColor": "#EEEFF1",
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      style={
        {
          "paddingRight": 36,
        }
      }
    >
      <Text
        allowFontScaling={true}
        style={
          {
            "color": "#5E6678",
            "fontFamily": "Neotis",
            "fontSize": 72,
          }
        }
      >
        Ó®ê
      </Text>
    </View>
    <Text
      allowFontScaling={true}
      style={
        [
          {
            "color": "#2B78EE",
            "fontFamily": "Inter-Medium",
            "fontSize": 48,
            "lineHeight": 60,
          },
          {
            "paddingVertical": 45,
          },
        ]
      }
    >
      title
    </Text>
  </View>
</View>
`;

exports[`@npm/mobydick-core/ActionSheetBase should renders correctly Item with selected 1`] = `
<View
  accessibilityLabel="accessibilityLabelActionSheetsItem"
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 36,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 24,
        "marginHorizontal": 24,
        "minHeight": 150,
        "paddingHorizontal": 60,
      },
      {
        "backgroundColor": "#FFF",
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <Text
      allowFontScaling={true}
      style={
        [
          {
            "color": "#20242D",
            "fontFamily": "Inter-Regular",
            "fontSize": 48,
            "lineHeight": 60,
          },
          {
            "paddingVertical": 45,
          },
        ]
      }
    >
      title
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "maxWidth": "100%",
          "opacity": 1,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderColor": "rgba(32, 36, 45, 0.3)",
          "borderRadius": 12,
          "borderWidth": 0,
          "height": 60,
          "width": 60,
        }
      }
      value="title"
    >
      <SvgMock
        height={60}
        width={60}
      />
    </View>
  </View>
</View>
`;

exports[`@npm/mobydick-core/ActionSheetBase should renders correctly inner item 1`] = `
<View
  accessibilityLabel="accessibilityLabelActionSheetsItem"
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderTopColor": "#D1D4DB",
        "borderTopWidth": 3,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": 24,
        "minHeight": 150,
        "paddingHorizontal": 60,
      },
      {
        "backgroundColor": "#FFF",
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <Text
      allowFontScaling={true}
      style={
        [
          {
            "color": "#FFF",
            "fontFamily": "Inter-Regular",
            "fontSize": 45,
            "lineHeight": 57,
          },
          {
            "paddingVertical": 45,
          },
        ]
      }
    >
      title
    </Text>
  </View>
</View>
`;
